src/helpers/path_ext.rs

```diff
@@ -0,0 +1,39 @@
+//! Shamelessly taken from:
```

Comment:
There are two external crates that do the same thing:
 - [`sugar_path::PathSugar`](https://docs.rs/sugar_path/latest/sugar_path/trait.PathSugar.html)
 - [`lexiclean::Lexiclean`](https://docs.rs/lexiclean/latest/lexiclean/trait.Lexiclean.html)

But I'm not sure how actively maintained they are and whether we should use an external crate for this.